python/action-server/main.py

```diff
@@ -0,0 +1,154 @@
+    def query_handler(self, query: Query):
+        print(">> [Queryable ] Received Query '{}'".format(query.selector))
+        replies = []
+        for stored_name, sample in store.items():
+            if query.key_expr.intersects(stored_name):
+                query.reply(sample)
+
+    # for publishing the data on subscriber or can be said that the to take the feedback
+    def publisher(self, key, pub, iter):
+        for idx in itertools.count() if iter is None else range(int(iter)):
+            time.sleep(1)
+            buf = (idx % 100) 
+            print(f"Putting Data ('{key}': '{buf}')...")
+            pub.put(buf)
+        
+
+class Session(Handlers):
+    # creates self.session and perform self.session related tasks. Takes an object of settings.
+    def __init__(self, settings):
+        self.setting = settings
+        self.session = None
+        self.sub = None
+        self.pub = None
+        self.queryable = None
+
+    # configures the zenoh configuration from settings variables and returns a configuration object.
+    def configuration(self):
+        conf = zenoh.Config.from_file(
+            self.setting.config) if self.setting.config is not None else zenoh.Config()
+        if self.setting.mode is not None:
+            conf.insert_json5(zenoh.config.MODE_KEY, json.dumps(self.setting.mode))
+        if self.setting.connect is not None:
+            conf.insert_json5(zenoh.config.CONNECT_KEY, json.dumps(self.setting.connect))
+        if self.setting.listen is not None:
+            conf.insert_json5(zenoh.config.LISTEN_KEY, json.dumps(self.setting.listen))
+        return conf
+
+    # defines target ofqueryable.
+    def target(self):
+        target = {
+            'ALL': QueryTarget.ALL(),
+            'BEST_MATCHING': QueryTarget.BEST_MATCHING(),
+            'ALL_COMPLETE': QueryTarget.ALL_COMPLETE(),}.get(self.setting.target)
+        return target
+
+    # sends the query toqueryable and print its output and returns a value of the sent keyexpr.
+    def get(self, end_expr):
+        result = self.session.get(self.setting.base_key_expr+end_expr, zenoh.ListCollector(), target=self.target())
+        for reply in result():
+            try:
+                key= reply.ok.key_expr
+                value =reply.ok.payload.decode("utf-8")
+            except:
+                raise_error(reply.err.payload.decode("utf-8"))
+        return value
+
+    # starts the action server and declares subscriqueryable and publisher.
+    def setup_action_server(self):
+        # initiate logging
+        log.info('Starting Action Server....')
+        zenoh.init_logger()
+        self.session = zenoh.open(self.configuration())
+        
+        self.sub = self.session.declare_subscriber(self.setting.declare_key_expr, self.listener, reliability=Reliability.RELIABLE())
+
+        self.queryable = self.session.declare_queryable(self.setting.declare_key_expr, self.query_handler)
```

Comment:
On both these queries, the action server should do something.

- In case of `/statechart` request, the statechart json should be responded with
- In case of `/trigger` request with the body `{ timestamp: “786786”, event: “TASK_ABORT” }, the endpoint should respond with { response_code: “ok”, message: “trigger accepted” } or { response_code: “rejected”, message: “trigger not recognized. trigger must be a valid trigger present in the statechart.” }, etc.

Reply:
Created two queryables.
Working on trigger request.

Reply:
Done with this. I have created two query handlers:
        1. query_handler for statechart which will return a json file of statechart.
        2. query_handler for trigger which will work as you have explained above.